File: 010/main2.c
#include <stdio.h>
#include <math.h>

/* int N = 10; */
int N = 2000000;

int main () {
    int p[N];
    p[0] = 2;
    int i = 1;
    for (int n = 3; n < N; n += 2) {
        int isprime = 1;
        for (int k = 0; k < i; k++) {
            if (p[k] > sqrt(n)) {
                break;
            }
            if (n % p[k] == 0) {
                isprime = 0;
                break;
            }
        }
        if (isprime == 1) {
            /* printf("%d\n", n); */
            p[i] = n;
            i++;
        }
    }
    long sum = 0;
    for (int k = 0; k < i; k++) {
        sum += p[k];
    }
    printf("%ld\n", sum);
    return 0;
}

File: 041/main2.py
#!/usr/bin/env python

from numpy import ones, sqrt

def ispandigital(n):
    """check if a number is pandigital"""
    s = str(n)
    l = len(s)
    s = s.replace('0', '')  # remove 0
    s = ''.join(sorted(set(s))) # remove duplicates
    if len(s) == l and int(s[-1]) == l:
        return True
    else:
        return False


def test_ispandigital():
    assert ispandigital(12304) == False
    assert ispandigital(1245) == False
    assert ispandigital(12245) == False
    assert ispandigital(123) == True

def isprime(n):
    """check if an odd number is prime"""
    for d in range(3, int(sqrt(n)) + 1):
        if n % d == 0:
            return False
    return True

def test_isprime():
    assert isprime(3) == True
    assert isprime(2143) == True
    assert isprime(9) == False


n = 9999999
while True:
    if isprime(n) and ispandigital(n):
        print n
        break
    n -= 2
